packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -0,0 +1,294 @@
+import { InstanceType, SubnetSelection } from '@aws-cdk/aws-ec2';
+import { IRole, ManagedPolicy, Role, ServicePrincipal } from '@aws-cdk/aws-iam';
+import { CfnOutput, Construct, IResource, Resource } from '@aws-cdk/core';
+import { Cluster } from './cluster';
+import { CfnNodegroup } from './eks.generated';
+
+/**
+ * NodeGroup interface
+ */
+export interface INodegroup extends IResource {
+  /**
+   * Name of the nodegroup
+   * @attribute
+   */
+  readonly nodegroupName: string
+}
+
+/**
+ * Nodegroup attributes used for the import
+ */
+export interface NodegroupAttributes {
+  /**
+   * Name of the nodegroup
+   */
+  readonly nodegroupName: string;
+}
+
+/**
+ * The AMI type for your node group. GPU instance types should use the `AL2_x86_64_GPU` AMI type, which uses the Amazon EKS-optimized
+ * Linux AMI with GPU support. Non-GPU instances should use the `AL2_x86_64` AMI type, which uses the Amazon EKS-optimized Linux AMI.
+ */
+export enum AmiType {
```

Comment:
```suggestion
export enum NodegroupAmiType {
```

(the namespace is for the entire EKS module)